pkg/build/builder/cmd/builder_test.go

```diff
@@ -0,0 +1,266 @@
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	buildv1 "github.com/openshift/api/build/v1"
+	"github.com/openshift/library-go/pkg/git"
+)
+
+func TestSetupProxyConfig(t *testing.T) {
+	fakeHTTPProxy := "http://proxy.io"
+	fakeHTTPSProxy := "https://proxy.io"
+
+	cases := []struct {
+		name                  string
+		expectError           bool
+		build                 *buildv1.Build
+		expectedLocalConfigs  map[string]string
+		expectedGlobalConfigs map[string]string
+	}{
+		{
+			name:  "no git",
+			build: &buildv1.Build{},
+		},
+		{
+			name: "git clone no proxy no auth",
+			build: &buildv1.Build{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-build-1",
+					Namespace: "test",
+				},
+				Spec: buildv1.BuildSpec{
+					CommonSpec: buildv1.CommonSpec{
+						Source: buildv1.BuildSource{
+							Type: buildv1.BuildSourceGit,
+							Git: &buildv1.GitBuildSource{
+								URI: "https://githost.dev/myorg/myrepo.git",
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "git clone proxy no auth",
+			build: &buildv1.Build{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-build-1",
+					Namespace: "test",
+				},
+				Spec: buildv1.BuildSpec{
+					CommonSpec: buildv1.CommonSpec{
+						Source: buildv1.BuildSource{
+							Type: buildv1.BuildSourceGit,
+							Git: &buildv1.GitBuildSource{
+								URI: "https://githost.dev/myorg/myrepo.git",
+								ProxyConfig: buildv1.ProxyConfig{
+									HTTPProxy:  &fakeHTTPProxy,
+									HTTPSProxy: &fakeHTTPSProxy,
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedGlobalConfigs: map[string]string{
+				"http.proxy":  fakeHTTPProxy,
+				"https.proxy": fakeHTTPSProxy,
+			},
+		},
+		{
+			name: "git clone proxy",
+			build: &buildv1.Build{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-build-1",
+					Namespace: "test",
+				},
+				Spec: buildv1.BuildSpec{
+					CommonSpec: buildv1.CommonSpec{
+						Source: buildv1.BuildSource{
+							Type: buildv1.BuildSourceGit,
+							Git: &buildv1.GitBuildSource{
+								URI: "https://githost.dev/myorg/myrepo.git",
+							},
+							SourceSecret: &corev1.LocalObjectReference{
+								Name: "git-auth",
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "git clone proxy auth",
+			build: &buildv1.Build{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-build-1",
+					Namespace: "test",
+				},
+				Spec: buildv1.BuildSpec{
+					CommonSpec: buildv1.CommonSpec{
+						Source: buildv1.BuildSource{
+							Type: buildv1.BuildSourceGit,
+							Git: &buildv1.GitBuildSource{
+								URI: "https://githost.dev/myorg/myrepo.git",
+								ProxyConfig: buildv1.ProxyConfig{
+									HTTPProxy:  &fakeHTTPProxy,
+									HTTPSProxy: &fakeHTTPSProxy,
+								},
+							},
+							SourceSecret: &corev1.LocalObjectReference{
+								Name: "git-auth",
+							},
+						},
+					},
+				},
+			},
+			expectedLocalConfigs: map[string]string{
```

Comment:
so I see us validating the proxy stuff is there, but maybe based on the 2 source secret types we support for the proxy scenario (userid pwd and included .gitconfig) we should verify that their content was not removed ... i.e. we are in fact updating the same file @adambkaplan 

so if we 
a) leverage the k8s fake secret client stuff for tests
b) create a secret that looked like 
    i) the basic auth style of secret fashioned from https://docs.openshift.com/container-platform/4.6/builds/creating-build-inputs.html#builds-source-secret-basic-auth_creating-build-inputs
   ii) the include your own .gitconfig file secret per https://docs.openshift.com/container-platform/4.6/builds/creating-build-inputs.html#builds-gitconfig-file-secured-git_creating-build-inputs

we would fully could fully validate that the new git use local config stuff updates / includes that stuff accordingly

In particular, consider the password case where it creates `gitcredentials.*` file to include in the "git config" file it names `gitcredentialscfg.`

let's make sure after your local call the same file has everything.

the user provided .gitconfig file case is probably more straight forward

WDYT ?

Reply:
IMO we don't have to go so far as to wire in a fake openshift client. We can probably cover the items you are talking about with a unit test for `setupGitEnvironment`. I'll include one to make sure that it is returning a path to a `.gitconfig` which has appropriate include clauses.

Reply:
I tried writing a good unit test for this function. Unfortunately there are side-effects which write files to temp dirs that are difficult to track. In general I don't like adding tests which can't clean themselves up.

Added https://issues.redhat.com/browse/BUILD-194 to address.

cc @coreydaley @otaviof